src/datatypes/numeric.js

```diff
@@ -9,8 +9,12 @@ function readI64 (buffer, offset) {
 }
 
 function writeI64 (value, buffer, offset) {
-  buffer.writeInt32BE(value[0], offset)
-  buffer.writeInt32BE(value[1], offset + 4)
+  if (typeof value === 'bigint') {
```

Comment:
what about reading?
This introduced an inconsistency between values read by i64 and the ones written by it

Reply:
Reading would be a breaking change, would also need to bump the node version to 10.4

Reply:
yes. I'm not sure it makes sense to change writer but not reader
How do you then use that?

Reply:
In PrismarineJS/prismarine-nbt#35 I have it reading 64-bit zigzag varints into a BigInt to keep things simple and prevent an overflow. So to make sure that the file can be written back in a non-varint format I added support for writing them here, but yeah it looks like it could cause API issues. Just thought it would be a bit wasteful to have to do extra conversions between number/bigint in the internal varint implementation

Reply:
okay so I've been looking at converting to arrays in pnbt but after thinking about it for a bit I can't really think of any good use case to send back arrays. If the user has to convert them to bigint to do anything useful with them (e.g. to read/update a timestamp) and then the user has to send back an array to pnbt, and then that has to be converted into a bigint, then that seems a bit excessive , especially as operations on big numbers in JS is already slow.